kernel/src/transaction.rs

```diff
@@ -143,13 +143,13 @@ impl Transaction {
     fn generate_logical_to_physical(&self) -> Expression {
         // for now, we just pass through all the columns except partition columns.
         // note this is _incorrect_ if table config deems we need partition columns.
-        let partition_columns = self.read_snapshot.metadata().partition_columns.clone();
+        let partition_columns = &self.read_snapshot.metadata().partition_columns;
         let fields = self.read_snapshot.schema().fields();
         let fields = fields.filter_map(|f| {
             if partition_columns.contains(f.name()) {
                 None
             } else {
-                Some(ColumnName::new([f.name()]).into())
+                Some(Expression::column([f.name()]))
             }
         });
```

Comment:
FYI: When I tried using `then(|| ...)` for this (which is equivalent), clippy suggests to use `filter` followed by `map`.
We should probably follow that advice here, since this code is even bulkier than `then` (but complex enough that clippy can't tell what's going on).

Reply:
oh interesting. ill do filter then map then :)